components/Browse.js

```diff
@@ -1,8 +1,46 @@
-const Browse = (props) => {
+const BrowseCard = (props = {}) => {
   return `
-    <div>
-    
+    <div class="twoD-browse-card" draggable dragid="inventory-${props.id}">
+      <img class="twoD-browse-card-preview" src="${props.preview}"></img>
+      <h4 class="twoD-browse-card-name">${props.name}</h4>
+      <div class="twoD-browse-card-actions">
+        <button class="twoD-browse-card-wearBtn" onclick=inventory-spawn name="${props.id}">
+          <i class="fal fa-magic" style="margin-right: 5px;"></i>
+          Spawn
+        </button>
+        <button class="twoD-browse-card-wearBtn" onclick=inventory-wear name="${props.id}">
+          <i class="fal fa-tshirt" style="margin-right: 5px;"></i>
+          Wear
+        </button>
+        <button class="twoD-browse-card-discardBtn" onclick=inventory-discard name="${props.id}">
+          <i class="fal fa-trash" style="margin-right: 5px;"></i>
+          Discard
+        </button>
+        <button class="twoD-browse-card-inspectBtn">
+          <i class="fal fa-search-plus" style="margin-right: 5px;"></i>
+          Inspect
+        </button>
+      </div>
     </div>
   `;
-}
+};
+const Browse = (props = {}) => {
+  console.log(props)
+  let allItems = props.allItems || [];
+  return `
+    <div class="twoD-browse">
+      <div class="twoD-browse-list">
+        ${
+          allItems.map((value, index) => {
+            return BrowseCard({
+              id: value.id,
+              name: value.filename || '',
```

Comment:
Can this ever be blank? If not it shouldn't have a default.

Reply:
could be blank if a bug or something

Reply:
But then the bug should be fixed or we should throw. What about value, can that be blank? What about id? Where does it end?

Reply:
If this is a legit case we should document why this is possible. If not then there shouldn't be a default.

Reply:
we should check that further up the logic not in the "dumb views", this is just to not error in any situation of using the view

Reply:
But id can't be blank?